webstruct/html_tokenizer.py

```diff
@@ -98,6 +100,8 @@ class HtmlTokenizer(object):
     sequence_encoder : object, optional
         Sequence encoder object. If not passed,
         :class:`~webstruct.sequence_encoding.IobEncoder` instance is created.
+    bilou : if True it will generate IOB2 tags using `IobEncoder` and translate
+        them to BILOU tags, False by default.
```

Comment:
👍 for putting this option to tokenizer, I think it is a right thing to do. 

However, I think it is better to have a string constant with an encoding name, to make it extensible. So I'd add `sequence_encoding` parameter, with 'IOB' as a default, and 'BILOU' alternative value. But: if there is a `sequence_encoding` argument, it is not clear what is `sequence_encoder` argument for, i.e. why should it be customizable.

So, let's say we add `sequence_encoding` argument with 2 string constants, and have separate code paths for these constants right in HtmlTokenizer class (as it is now). If someone wants to add yet another scheme, HtmlTokenizer should be changed. This is not a big deal, but the original idea of `sequence_encoder` objects was to put all logic which is different between different encodings like IOB and BILOU to these classes.

I think we should only have one of `sequence_encoder` and `sequence_encoding` parameters.

Currently there are indeed two code paths in HtmlTokenizer: for bilou and for iob. It looks easy enough to move all encoding-dependent logic to the encoder classes, and just call necessary methods, so my vote is to drop `bilou` argument, and provide BilouEncoder object, as originally planned. I'll make some comments below.

Reply:
ok, so your idea would be to create a class BilouEncoder which would first encode tokens using IobEncoder and then would translate the labels to Bilou using the methods I already implemented, is this correct? 

And with 
>move all encoding-dependent logic to the encoder classes, and just call necessary methods
 
do yuou mean to move ``html_tokenizer._encode()`` into ``IobEncoder`` and ``BilouEncoder``?